pyroute2/netlink/__init__.py

```diff
@@ -1823,6 +1823,36 @@ def decode(self):
             nla_base.decode(self)
             self.value = ':'.join('%02x' % (i) for i in
                                   struct.unpack('BBBBBB', self['value']))
+            
+    class l2orv4(nla_base):
+        '''
+        Decode MAC or IPv4 address.
+        '''
+
+        __slots__ = ()
+        sql_type = 'TEXT'
+        fields = [('value', 's')]
+
+        def encode(self):
+            if self.value.find(':') > -1:
+                self['value'] = struct.pack('BBBBBB',
+                                            *[int(i, 16) for i in 
+                                              self.value.split(':')])
+            elif self.value.find('.') > -1:
```

Comment:
In my opinion, decoding values based on length is OK. But for encoding, we should wait explicit values from user, and not guessing values. Since encoding is done in python, we could set the type somewhere and use the proper function to encode data